clippy_lints/src/unwrap.rs

```diff
@@ -210,6 +210,11 @@ impl<'a, 'tcx> Visitor<'tcx> for UnwrappableVariablesVisitor<'a, 'tcx> {
 declare_lint_pass!(Unwrap => [PANICKING_UNWRAP, UNNECESSARY_UNWRAP]);
 
 impl<'tcx> LateLintPass<'tcx> for Unwrap {
+    fn check_item(&mut self, cx: &LateContext<'_>, item: &Item<'_>) {
+        if is_test_module_or_function(item) {
+            return;
+        }
+    }
```

Comment:
This won't work, since the underlying visitor behind the linting system will call the `check_fn` method anyway. See the `clippy_lints::wildcard_imports` lint on how to do this properly. (It involves the `check_item_post` method)